### PR TITLE
Nested Python API Refactor

### PR DESCRIPTION
**Pull request by JesseMckinzie**
-Refactored Nested class in the Python API
-Add documentation for Nested 
-Add example of new Nested api to the README

### CODE REVIEW

File: src/nyx/python/nested_roi_py.cpp [hunk 1]
```diff
@@ -659,5 +659,103 @@ bool mine_segment_relations (
...
+			if (!ok)
+				throw std::runtime_error("Error creating relational table of segments");
+		}
+
+		// Aggregate features
+		if (output2python)
+		{
+			// Aggregating is implementing externally to this function
+		}
+		else
```
[the remark this answers is not in the document]
Comment (JesseMckinzie) [reply src/nyx/python/nested_roi_py.cpp:751]: Yes, we do this so the aggregate functions provided by Pandas (or lambdas provided by the user) can be used on the python side.

File: src/nyx/python/new_bindings_py.cpp [hunk 2]
```diff
@@ -16,6 +16,16 @@ using namespace Nyxus;
 // Defined in nested.cpp
 bool mine_segment_relations(bool output2python, const std::string& label_dir, const std::string& file_pattern, const std::string& channel_signature, const int parent_channel, const int child_channel, const std::string& outdir, const ChildFeatureAggregation& aggr, int verbosity_level);
```
[the remark this answers is not in the document]
Comment (JesseMckinzie) [reply src/nyx/python/new_bindings_py.cpp:17]: Removed.

File: docs/source/examples.rst [hunk 3]
```diff
@@ -107,3 +107,159 @@ and the file is passed to Nyxus via parameter --intSegMapFile, the mapping will
...
+---------
+
+To use the Nested class, first call the constructor with the optional argument `aggregate`. If `aggregate` is not passed, the 
+`find_relation` behavior will change (described later). Any aggregate function supported by Pandas is available, 
+such as `min`, `max`, `count`, and `mean`. Lambda functions can also be used, and named using a 2-tuple, where the first 
+element is the name and the second is the lambda function. This allows functions that are not supported by Pandas to be used,
+such as Numpy's `np.nanmean`. 
+
+To use the Nested class, first call Nyxus to get the features of all ROIs from the child channels. If the child channels are described 
+by a channel number in the filename, a filepattern can be used to filter down to only the child channehl. Consider a directory with the images
```
[the remark this answers is not in the document]
Comment (JesseMckinzie) [reply docs/source/examples.rst:170]: Fixed.

File: docs/source/examples.rst [hunk 4]
```diff
@@ -107,3 +107,159 @@ and the file is passed to Nyxus via parameter --intSegMapFile, the mapping will
...
+   Image              Parent_Label  Child_Label
+   0  /path/to/image          72.0         65.0
+   1  /path/to/image          71.0         66.0
+   2  /path/to/image          70.0         64.0
+   3  /path/to/image          68.0         61.0
+   4  /path/to/image          67.0         65.0
+
+The `featurize` method can then be used along with the child features to apply the aggregate functions. The `featurize` method 
+takes in the `features` DataFrame generated by Nyxus, which contains the features calculations for each ROI, along with the DataFrame 
+containing the parent-child relations from the `find_relations` method. The ouput of this method is a DataFrame containing
```
[the remark this answers is not in the document]
Comment (JesseMckinzie) [reply docs/source/examples.rst:236]: Fixed.

File: README.md [hunk 5]
```diff
@@ -311,14 +311,97 @@ __Example__: we need to process collection of mask images located in directory "
...
 
-Using the hierarchical ROI Python API is illustrated in the following example:
+### Nested features Python API
+
+The nested features functionality can also be utilized in Python using the `Nested` class in `nyxus`. The `Nested` class
+contains two methods, `find_relations` and `featurize`. 
+
+The `find_relations` method takes in a path to the label files, along with a child 
+filepattern to identify the files in the child channel and a parent filepattern to match the files in the parent channel. The `find_relation` method 
+returns a Pandas DataFrame containing a mapping between parent ROIS and the respective child ROIs.
```
[the remark this answers is not in the document]
Comment (JesseMckinzie) [reply README.md:334]: Fixed.

**Review by sameeul — APPROVED** [review]

**Pull request merged**